src/seedu/addressbook/logic/Logic.java

```diff
@@ -85,7 +85,9 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        if(command.isMutating()){
+        	storage.save(addressBook);
+        }
```

Comment:
The behavior of the code has been changed. Ensure the header comment and the tests have been updated.

Coding standard violation